www/source/blog/2017-10-27-distributed-java-application.html.md

```diff
@@ -0,0 +1,360 @@
+```
+
+By using the `--bind` arg I am telling the supervisor which service it can expect to fulfill the provider side of the required binding contract. If for some reason the Supervisor did not find my mongodb service, then it would wait to start my service until it found one. This is one way we can enforce the order in which services are started in a Habitat ecosystem. BUT it DID find one and my service starts. By running `sup-log` in the Studio it will follow the tail of the Supervisor log which gives me more detailed information on what is happening. Here is just a sampling:
+
+```
+national-parks.default(O): 27-Oct-2017 19:16:44.107 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory /hab/pkgs/core/tomcat8/8.5.9/20170514144202/tc/webapps/host-manager has finished in 27 ms
+national-parks.default(O): 27-Oct-2017 19:16:44.107 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory /hab/pkgs/core/tomcat8/8.5.9/20170514144202/tc/webapps/docs
+national-parks.default(O): 27-Oct-2017 19:16:44.134 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory /hab/pkgs/core/tomcat8/8.5.9/20170514144202/tc/webapps/docs has finished in 27 ms
+national-parks.default(O): 27-Oct-2017 19:16:44.140 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler [http-nio-8080]
+national-parks.default(O): 27-Oct-2017 19:16:44.154 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler [ajp-nio-8009]
+national-parks.default(O): 27-Oct-2017 19:16:44.154 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in 2000 ms
+```
+
+If that's not some Tomcaty Java goodness then I don't know what is. Ladies and gentleman, I do believe a JVM is afoot.
+
+Lets check the ultimate authority: the browser:
+
+![](media/2017-10-27-modernize-java/parks.png)
+
+Now those are some national parks!!
+
+## Deploying to Azure
+
+So now that I am more than satisfied with how this renders locally, its time to share it with the world. Lets take a peek at my sweet Azure infra:
+
+![](media/2017-10-27-modernize-java/azure.png)
+
+I have three nodes running my Java Tomcat service: `hab1`, `hab2`, and `hab3`. I have `hab-mongo` running my mongodb service and lastly `hab-ha` is running my Habitat [`core/haproxy`](https://github.com/habitat-sh/core-plans/tree/master/haproxy) package and load balancing hab1 to hab3. It is the only node externally exposing port 8080 and what I expect external traffic to hit directly.
+
+I have the Supervisors on hab1 to hab3 configured with an [update strategy](https://www.habitat.sh/docs/using-habitat/#using-updates) set to `rolling`. In fact I "loaded" the Supervisors with this command:
+
+```
+sudo hab sup load --channel unstable mwrock/national-parks --bind databa
```

Comment:
The syntax coloring here is confusing-looking because the args are being rendered as comments. Is there a way to tell it to render as a command line?